subprojects/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultGradleEnterprisePluginCheckInService.java

```diff
@@ -87,9 +100,24 @@ public GradleEnterprisePluginServiceRef getPluginServiceRef() {
         };
     }
 
-    private boolean isUnsupportedWithConfigurationCaching(GradleEnterprisePluginMetadata pluginMetadata) {
-        VersionNumber version = VersionNumber.parse(pluginMetadata.getVersion()).getBaseVersion();
-        return isConfigurationCacheEnabled && MINIMUM_SUPPORTED_PLUGIN_VERSION_FOR_CONFIGURATION_CACHING.compareTo(version) > 0;
+    private boolean isUnsupportedWithConfigurationCaching(VersionNumber pluginBaseVersion) {
+        return isConfigurationCacheEnabled && MINIMUM_SUPPORTED_PLUGIN_VERSION_FOR_CONFIGURATION_CACHING.compareTo(pluginBaseVersion) > 0;
     }
 
+    private static boolean isDeprecatedPluginVersion(VersionNumber pluginBaseVersion) {
+        return MINIMUM_SUPPORTED_PLUGIN_VERSION_SINCE_GRADLE_9.compareTo(pluginBaseVersion) > 0;
+    }
+
+    private static void nagAboutDeprecatedPluginVersion() {
+        DeprecationLogger.deprecate("Usage of this Gradle Enterprise plugin version")
+            .withContext("The plugin application will be ignored.")
+            .withAdvice(String.format("Please upgrade to version %s.%s.%s or later of the Gradle Enterprise plugin.",
+                MINIMUM_SUPPORTED_PLUGIN_VERSION_SINCE_GRADLE_9.getMajor(),
+                MINIMUM_SUPPORTED_PLUGIN_VERSION_SINCE_GRADLE_9.getMinor(),
+                MINIMUM_SUPPORTED_PLUGIN_VERSION_SINCE_GRADLE_9.getMicro()
+            ))
+            .willChangeInGradle9()
+            .withUpgradeGuideSection(8, "unsupported_ge_plugin_3.13")
+            .nagUser();
```

Comment:
❓ Does the method return an indirect or a direct usage of the deprecation? I think in this case the usage should be indirect, since the stacktrace here isn't interesting for the user.

Reply:
I didn't know about this distinction before. You are right, we should not display the stacktrace. I changed the deprecation call to `DeprecationLogger.deprecateIndirectUsage`, which omits the stacktrace in the build scan:

<img width="752" alt="image" src="https://github.com/gradle/gradle/assets/2759152/edc5d555-fb72-461d-abcf-d1698380f755">